core/issuers.js

```diff
@@ -17,16 +17,11 @@ Description must be truthy: ${description}`;
 
   // src is a purse or payment. Return a fresh payment.  One internal
   // function used for both cases, since they are so similar.
-  function takePayment(amount, isPurse, src, _name) {
```

Comment:
Take a look at `issuer.getExclusive`. Without the `isPurse` flag, what prevents `getExclusive` from succeeding if passed a purse?

Reply:
Hi Mark, `isPurse` as a parameter doesn't solve this problem. I agree that `getExclusive` should error if a purse is passed in rather than a payment, but if you look at the [original code for getExclusive](https://github.com/Agoric/ERTP/blob/d577bd40a1f7061cb9cc78cc44a91ac6c31fdc51/core/issuers.js#L84), it passes in `false` for `isPurse` regardless, meaning that it will succeed even when a purse is actually passed in. I think we can find a solution to the problem, but in the meantime, I think we should take `isPurse` out.

Reply:
Looked. Called from `getExclusive` it is always false, so only payments are accepted. Called from `withdraw`, it is always true, so only purses are accepted.

Reply:
Mark, can you explain how this prevents a payment from actually being passed in in place of a purse and vice versa?

Reply:
In the original you link to, at https://github.com/Agoric/ERTP/blob/d577bd40a1f7061cb9cc78cc44a91ac6c31fdc51/core/issuers.js#L36 is the code

```js
    if (isPurse) {
      insist(useRights.has(src))`\
Purse expected: ${src}`;
    } else {
      insist(homePurses.has(src))`\
Payment expected: ${src}`;
    }
```

In that original, only purses were registered in `useRights`. Only payments were registered in `homePurses`. Of course, when we lost `useRights` and `homePurse` we lost that too, so we need another way to distinguish these. The remaining PrivateName, `rights`, has been abstracted into MintController, whose API names also claim to distinguish between purses and payments, but use one `rights` table for both.

I think the cleanest way to fix this is for MintController to have two PrivateNames, `purseRights` and `paymentRights`, so that only purses are found in the first and only payments are found in the second. Avoid using one brand to certify two different things.

Reply:
Sure, sounds good. I'll add that

Reply:
That's added. @Chris-Hibbert  I had to make some changes to the corkboard tests [here](https://github.com/Agoric/ERTP/blob/4776c4c3a555e05d236f5608d627f918f0198402/demo/gallery/vat-bob.js#L72) because sometimes purses were being passed as payments and vice versa.